src/useValidator/__docs__/example.stories.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { useState } from 'react';
+import { IValidatorDeferred, useDebounceCallback, useValidator } from '../..';
+
+export const ExampleStories: React.FC = () => {
+  const [text, setText] = useState('');
+
+  // as deferred validator is unable to infer the type of validity
+  // state - we should define it ourself
+  type ITextValidityState = { isValid: boolean | undefined; error: Error | undefined };
+
+  // debounced callback is deferred callback so we should use deferred type
+  // of validator (the one that receives dispatcher as an argument)
+  const validator = useDebounceCallback<IValidatorDeferred<ITextValidityState>>(
+    (d) => {
+      const isValid = !text.length || text.length % 2 === 1;
+
+      d({
+        isValid,
+        error: !isValid ? new Error('text length should be an odd length') : undefined,
+      });
+    },
+    150,
+    [text]
+  );
+
+  // validity state type if inferred from validator
+  const [validity] = useValidator(validator, [validator]);
```

Comment:
I don't know how I feel about this one. It almost feels like the better UX here would to be to either make this into useValidatorState, or to at least pass the value you are validating to useValidator.

Also, how it's set up here makes it so useValidator seems to be be pretty useless if you weren't using `useDebounceCallback`. You could otherwise just validate on input or on render, which both seem like better practices. Maybe I'm missing something though.

Reply:
> to either make this into useValidatorState
It will be implemented as a separate hook later.

> at least pass the value you are validating to useValidator
This will strict you to use validated value only, but what if you want to validate multiple values simultaneously? How to pass them to the validator? Dependencies list is an array, attempt to determine values depending on positions can cause inambiguos bugs.

> seems to be be pretty useless if you weren't using useDebounceCallback.
Althoug it supports synchronous validation - it was designed mostly for async\deferred validation.

> You could otherwise just validate on input or on render, which both seem like better practices.
On-input validation makes multivalue validation very complicated complicated.  
On-render validation forces you to create validity storage by hand and does validation on ach render instead of on-change.

Reply:
`it was designed mostly for async\deferred validation.`

How about we rename this hook to `useAsyncValidator` then? Or `useValidatorEffect`? I think that'd convey the intended use case better.

Reply:
It is state, not the effect, also it is not the async - only, as for my code I have several synchronous validators.

Reply:
Gotcha. I'm okay rolling with this for now, but I'm still not sold on it.